### ISSUE TITLE
DefaultBinderFactory defaultDomain retrieval by name fails

### ISSUE THREAD

**Issue by kesyu** [spring-cloud/spring-cloud-stream#1408]
Just a small thing that you might want to know about, located in
https://github.com/spring-cloud/spring-cloud-stream/blob/dae35f2dbf9ed8b2881b2b9e4ed4a66e798d6367/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java#L210

**Comment by olegz** [comment]
Sorry, I am lost, what's incompatible? Could you please provide some details?

**Comment by kesyu** [comment]
Sorry, let me rephrase that. That line, which I referenced, is not able to find the `spring.jmx.default-domain: myDomain` property in my bootstrap.yml.

Just did a quick re-check, and it's not really a snakeYaml issue, it's that in the `environment.getProperty("spring.jmx.default-domain.")`, the name has an extra dot at the end.

The property is found if I use `environment.getProperty("spring.jmx.default-domain")`

**Comment by olegz** [comment]
Ohh, good catch! Thank you!
